lib/service/Service.ts

```diff
@@ -170,7 +170,32 @@ class Service extends EventEmitter {
     argChecks.HAS_HOST({ host });
     argChecks.VALID_PORT({ port });
 
+    const peerActive = new Promise((resolve) => {
+      this.pool.once('peer.active', (peerPubKey) => {
+        if (peerPubKey === nodePubKey) {
+          resolve();
+        }
+      });
+    });
+
+    const peerClose: Promise<string> = new Promise((resolve) => {
+      this.pool.once('peer.close', (peerPubKey, rejectionMsg) => {
+        if (peerPubKey === nodePubKey) {
+          resolve(rejectionMsg);
+        }
+      });
+    });
+
+    const timeout: Promise<void> = new Promise((resolve) => {
+      setTimeout(resolve, 1000);
+    });
+
     await this.pool.addOutbound({ host, port }, nodePubKey, retryConnecting, true);
+    await peerActive;
+    const rejectionMsg = await Promise.race([peerClose, timeout]);
+    if (rejectionMsg !== undefined) {
```

Comment:
Wouldn't rejectionMsg be `undefined` if it times out? This could possibly be solved as well if you go with a single promise which rejects on timeout as I suggested above.

Reply:
When I put `setTimeout` into the `new Promise`, it will be executed immediately. And if addOutbound or peerActive blocks then timeout reject will be triggered first. And that timeout is  not the time waiting close packet but the time waiting addOutbound, peerActive, peerClose all together I think.

Reply:
Why would it be executed immediately? I realize that the time to complete the handshake may actually take longer than our timeout limit, but that's going to be an issue regardless of where the timeout is in this method.